python/ambassador/config/resourcefetcher.py

```diff
@@ -233,16 +227,16 @@ def parse_watt(self, serialization: str, finalize: bool=True) -> None:
 
             watt_k8s = watt_dict.get('Kubernetes', {})
 
-            # Handle normal Kube objects... the order is important here as ingresses depend on ingressclasses
-            for key in [ 'service', 'endpoints', 'secret', 'ingressclasses', 'ingresses' ]:
-                for obj in watt_k8s.get(key) or []:
-                    # self.logger.debug(f"Handling Kubernetes {key}...")
-                    self.handle_k8s(obj)
+            # These objects have to be processed first, in order, as they depend
+            # on each other.
+            watt_k8s_keys = ['service', 'endpoints', 'secret', 'ingressclasses', 'ingresses']
 
-            # ...then handle Ambassador CRDs.
-            for key in CRDTypes:
+            # Then we add everything else to be processed.
+            watt_k8s_keys += watt_k8s.keys()
+
+            for key in dict.fromkeys(watt_k8s_keys):
```

Comment:
Hmmmmm. So your comment on line 230 for `watt_k8s_keys` lists a sequence of keys that must be processed _in order_, then this line generates a dict (to deduplicate the keys, presumably)... so won't iterating over the dict shred the order you so carefully set up?

Reply:
I will add a comment to clarify, but this is a basically a convenient way to get around the lack of a distinct `OrderedSet` type. As of [Python 3.7](https://mail.python.org/pipermail/python-dev/2017-December/151283.html), dictionaries are insertion-ordered:

```
Python 3.7.6 (default, Jan 19 2020, 22:34:52) 
[GCC 9.2.1 20200117] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> foo = ['bar', 'baz', 'quux', 'quux', 'baz']
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
>>> dict.fromkeys(foo)
{'bar': None, 'baz': None, 'quux': None}
```